packages/eyeglass/src/assets/Assets.ts

```diff
@@ -1,242 +1,275 @@
-  var uri = new URI(originalUri);
-
-  // normalize the uri and resolve it
-
-  var data = resolveAssetDefaults.call(this, $assetsMap, uri.getPath());
-  if (data) {
-    var filepath = URI.restore(data.coerce.get("filepath"));
-
-    // create the URI
-    var fullUri = URI.join(
-      options.httpRoot,
-      options.assets.httpPrefix,
-      data.coerce.get("uri")
-    );
-
-    assets.resolve(filepath, fullUri, function(error, assetInfo) {
-      if (error) {
-        cb(error);
-      } else {
-        // if relativeTo is set
-        if (options.assets.relativeTo) {
-          // make the URI relative to the httpRoot + relativeTo path
-          uri.setPath(path.relative(
-            URI.join(options.httpRoot, options.assets.relativeTo),
-            assetInfo.path
-          ));
-        } else {
-          // otherwise, just update it to the path as is
-          uri.setPath(assetInfo.path);
-        }
-        // if a query param was specified, append it to the uri query
-        if (assetInfo.query) {
-          uri.addQuery(assetInfo.query);
-        }
+type ResolverCallback = (error: Error | null, result: Resolution | undefined) => any;
+type Resolver = (assetFile: string, assetUri: string, cb: ResolverCallback) => void;
+type WrappedResolver = (assetFile: string, assetUri: string, fallback: Resolver, cb: ResolverCallback) => void;
+
+interface Resolves {
+  resolve: Resolver;
+}
+
+type InstallerCallback = (error: Error | null, dest?: string) => void;
+type Installer = (file: string, uri: string, cb: InstallerCallback) => void;
+type WrappedInstaller = (file: string, uri: string, fallback: Installer, cb: InstallerCallback) => void;
+interface Installs {
+  install: Installer;
+}
+
+export default class Assets implements Resolves, Installs {
+  sassUtils: any;
+  eyeglass: IEyeglass;
+  /**
+   * Assets declared by the application.
+   */
+  collection: AssetsCollection;
+  /**
+   * Assets declared by eyeglass modules.
+   */
+  moduleCollections: Array<AssetsCollection>;
+  AssetCollection: () => AssetsCollection;
+  AssetPathEntry: (src: string, options: AssetSourceOptions) => AssetsSource;
+  constructor(eyeglass: IEyeglass, sass: SassImplementation) {
+    this.sassUtils = require("node-sass-utils")(sass);
+    this.eyeglass = eyeglass;
+    // create a master collection
+    this.collection = new AssetsCollection(eyeglass.options);
+    // and keep a list of module collections
+    this.moduleCollections = [];
+
+    // Expose these temporarily for back-compat reasons
+    function deprecate(method: string) {
+      eyeglass.deprecate("0.8.3", "0.9.0", [
+        "The assets." + method + " interface will be removed from the public API.",
+        "If you currently use this method, please open an issue at",
+        "https://github.com/sass-eyeglass/eyeglass/issues/ so we can",
+        "understand and accommodate your use case"
+      ].join(" "));
+    }
+    this.AssetCollection = function () {
```

Comment:
No reason other than it was already done this way and I've tried to take a "minimal impact" approach in this PR. Since this is deprecated code, I am more concerned with removing it than cleaning it up. I'd like to issue a 2.0 release where we remove all the deprecated APIs and types. It will really clean up the code. https://github.com/sass-eyeglass/eyeglass/issues/199